code-challenges.test.js

```diff
@@ -11,51 +11,272 @@
+// I will create an array that holds the string that is all lowercase
+// as I iterate over the array I will change the letters to the numbers
+//My funciton will then return a new array 
+//I will then change the array back into a string 
+//I attempted to refactor with ternary operators to cut down on the amount of code but didn't have much luck. 
+// I also tried to use only one .toLowerCase method to blanket all the strings but realized it doesn't work when returning the string (haha) so I put them in each conditional statement.
+
+const secretCoder = (string) =>{
+    let arrOne = string.split("")
+    arrTwo = arrOne.map((value, index) => {
+    if(value.toLowerCase() === "a"){
+        return 4
+    }
+    else if(value.toLowerCase() === "e"){
+        return 3
+    }
+        else if(value.toLowerCase() == "i"){
+            return 1
+        }  
+            else if(value.toLowerCase() == "o"){
+                return 0
+            }  
+            return value
+            
+            })
+            return arrTwo.join("")
+            }
+
+
+
 
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
+// const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+// const letterA = "a"
+// // Expected output: ["Apple", "Banana", "Orange"]
+// const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+// const letterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+//My function is going to be named extractor
+
+describe("extractor", () => {
 const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
 const letterA = "a"
 // Expected output: ["Apple", "Banana", "Orange"]
 const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
 const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
+it("takes in an array of words and a single letter and returns all the words that contain that particular letter ", () => {
+  expect(extractor(arrayOfWords1,letterA)).toEqual(["Apple", "Banana", "Orange"])
+  expect(extractor(arrayOfWords2,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+})
+})
 
-
+//ran test and failed -> function was not defined
 
 // b) Create the function that makes the test pass.
 
+//First I will need to create a funciton
+//the function will take 2 arguments, an array and a letter.
+//I will then iterate over the array to find words that contain a specific letter
+//I will then return the words containing the specified letter in an array
+//I will also need to find a way to make sure that I get both the uppercase and lowercase of the specified letter.
+
+const extractor = (array,letter) => {
+    extractdArr = array.filter(value => {
+      return value.toLowerCase().includes(letter)
+    })
+    return extractdArr
+    }
+
```

Comment:
Looks great to me. You might try applying the `.toLowerCase()` as well.
`return value.toLowerCase().includes(letter.toLowerCase())`